### PR TITLE
[Shopify] Pre-validate item before export in Add to Shopify action

### PR DESCRIPTION
**Pull request by onbuyuka**
## Summary
- Add `CheckItemCanBeExported` and `CheckItemVariantCount` to `ShpfyProductExport` to validate items before export (blocked, empty description, >2048 variants)
- Call pre-validation from the Item Card "Add to Shopify" action, preventing the misleading confirmation-then-silent-skip flow
- Consolidate duplicate checks: the `Add Item to Shopify` report and `CreateTempProduct` now call the shared validation instead of maintaining their own copies
- Add 3 TestPage-based tests covering blocked items, empty descriptions, and too-many-variants scenarios from the Item Card

Fixes [AB#626637](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/626637)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/Apps/W1/Shopify/App/src/Products/Codeunits/ShpfyProductExport.Codeunit.al [hunk 1]
```diff
@@ -921,8 +921,64 @@ codeunit 30178 "Shpfy Product Export"
...
+            SkippedRecord.LogSkippedRecord(Item.RecordId, ItemIsBlockedOrSalesBlockedLbl, Shop);
+            exit(false);
+        end;
+
+        if Item.Description = '' then begin
+            SkippedRecord.LogSkippedRecord(Item.RecordId, ItemDescriptionIsEmptyLbl, Shop);
+            exit(false);
+        end;
+
+        if not CheckItemVariantCount(Item) then
```
Comment (Groenbech96) [inline src/Apps/W1/Shopify/App/src/Products/Codeunits/ShpfyProductExport.Codeunit.al]: Return the function

File: src/Apps/W1/Shopify/App/src/Products/Codeunits/ShpfyProductExport.Codeunit.al [hunk 2]
```diff
@@ -921,8 +921,64 @@ codeunit 30178 "Shpfy Product Export"
...
+            exit(false);
+
+        exit(true);
```
Comment (onbuyuka) [inline src/Apps/W1/Shopify/App/src/Products/Codeunits/ShpfyProductExport.Codeunit.al]:
```suggestion
        exit(CheckItemVariantCount(Item));
```

File: src/Apps/W1/Shopify/App/src/Products/Page Extensions/ShpfyItemCard.PageExt.al [hunk 3]
```diff
@@ -53,6 +53,8 @@ pageextension 30119 "Shpfy Item Card" extends "Item Card"
                     begin
                         if SyncProducts.ConfirmAddItemToShopify(Rec, Shop) then begin
                             ProductExport.SetShop(Shop);
+                            if not ProductExport.CheckItemCanBeExported(Rec) then
+                                exit;
                             if not ProductExport.CheckItemAttributesCompatibleForProductOptions(Rec) then
                                 exit;
```
Comment (Groenbech96) [inline src/Apps/W1/Shopify/App/src/Products/Page Extensions/ShpfyItemCard.PageExt.al:59]: Merge together.

**Review by Groenbech96 — APPROVED** [review]

File: src/Apps/W1/Shopify/App/src/Products/Page Extensions/ShpfyItemCard.PageExt.al (hunk 3 above)
Comment (onbuyuka) [reply src/Apps/W1/Shopify/App/src/Products/Page Extensions/ShpfyItemCard.PageExt.al:59]: won't fix
